### PR TITLE
fix #227

### LINKED ISSUE #227: `row_max_abs_value` Causes Segfault
When building with RAJA and OpenMP (`HIOP_USE_GPU=OFF`) on tag v0.4.3 I find an invalid memory access in sparse matrix tests, seems to also be in `hiopMatrixRajaSparseTriplet::row_max_abs_value`:

```shell
$ mpirun -n 1 ./tests/testMatrixSparse
Testing hiopMatrixSparseTriplet
// these all pass

Testing hiopMatrixRajaSparseTriplet
// they pass up to here:
--- PASS: Test matrixTimesMatTrans
*** Error in `./tests/testMatrixSparse': double free or corruption (out): 0x0000000000b7a170 ***
--- PASS: Test matrixAddMDinvNtransToSymDeMatUTri
```

This is a bit misleading, since the actual segfault occurs in line 187 of `tests/testMatrixSparse.cpp`. Valgrind indicates this is caused by `hiopMatrixRajaSparseTriplet::row_max_abs_value` in `src/LinAlg/hiopMatrixRajaSparseTriplet.cpp:515`.

Valgrind:
```
==14243== Thread 17:
==14243== Invalid read of size 4
==14243==    at 0x460B89: hiop::hiopMatrixRajaSparseTriplet::row_max_abs_value(hiop::hiopVector&)::{lambda(long)#1}::operator()(long) const (hiopMatrixRajaSparseTriplet.cpp:515)
==14243==    by 0x468447: forall_impl<RAJA::TypedRangeSegment<long int>&, hiop::hiopMatrixRajaSparseTriplet::row_max_abs_value(hiop::hiopVector&)::<lambda(RAJA::Index_type)>&> (forall.hpp:89)
```

At this point in the problematic routine:
```cpp
 511│   RAJA::forall<hiop_raja_exec>(
 512│     RAJA::RangeSegment(0, num_rows+1),
 513│     RAJA_LAMBDA(RAJA::Index_type row_id)
 514│     {
 515│       for(int itnz=idx_start[row_id]; itnz<idx_start[row_id+1]; itnz++) {
 516│         double abs_val = fabs(values[itnz]);
 517│         vd[row_id] = (vd[row_id] > abs_val) ? vd[row_id] : abs_val;
 518│       }
 519│     }
 520├─> );
 521│ }
```

I find the following values in the sparse matrix in GDB:
```
nrows_ == 5
ncols_ == 50
values == {1 <repeats 23 times>} (is this the issue? Seems to be an array of len 23)
row_starts_host->num_rows_ == 5
row_starts_host->idx_start_ == {0, 5, 10, 15, 20}
```

It seems to me the invalid read occurs here:
```cpp
 511│   RAJA::forall<hiop_raja_exec>(
                                       // v---- Should this really be num_rows+1?
 512│     RAJA::RangeSegment(0, num_rows+1),
 513│     RAJA_LAMBDA(RAJA::Index_type row_id)
 514│     {                                                  // v---- right here
 515│       for(int itnz=idx_start[row_id]; itnz<idx_start[row_id+1]; itnz++) {
 516│         double abs_val = fabs(values[itnz]);
 517│         vd[row_id] = (vd[row_id] > abs_val) ? vd[row_id] : abs_val;
 518│       }
 519│     }
 520├─> );
 521│ }
```

CC @pelesh @cnpetra @nychiang

### PR DESCRIPTION
**Pull request by nychiang**
Fix a bug in RAJA implementation of `row_max_abs_value`.

CLOSE: #227

### CODE REVIEW

**Comment by nychiang** [comment]
This is a quick fix. Will merge it into `develop` once I receive one approval.

**Review by cnpetra — APPROVED** [review]

**Pull request merged**
